rocAL/rocAL/source/augmentations/geometry_augmentations/node_resize_mirror_normalize.cpp

```diff
@@ -0,0 +1,137 @@
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include <vx_ext_rpp.h>
+#include <graph.h>
+#include <cmath>
+#include "node_resize_mirror_normalize.h"
+#include "exception.h"
+
+ResizeMirrorNormalizeNode::ResizeMirrorNormalizeNode(const std::vector<Image *> &inputs, const std::vector<Image *> &outputs) :
+        Node(inputs, outputs), _mirror(MIRROR_RANGE[0], MIRROR_RANGE[1])
+{
+}
+
+void ResizeMirrorNormalizeNode::create_node()
+{
+    if(_node)
+        return;
+
+    std::vector<vx_float32> mean_vx, std_dev_vx;
+    _dest_width_val.resize(_batch_size);
+    _dest_height_val.resize(_batch_size);
+    mean_vx.resize(_batch_size * 3);
+    std_dev_vx.resize(_batch_size * 3);
+    for (uint i = 0; i < _batch_size; i++) {
+        mean_vx[3 * i] = _mean[0];
+        mean_vx[3 * i + 1] = _mean[1];
+        mean_vx[3 * i + 2] = _mean[2];
+
+        std_dev_vx[3 * i] = _std_dev[0];
+        std_dev_vx[3 * i + 1] = _std_dev[1];
+        std_dev_vx[3 * i + 2] = _std_dev[2];
+    }
+    _mean_array = vxCreateArray(vxGetContext((vx_reference)_graph->get()), VX_TYPE_FLOAT32, _batch_size * 3);
+    _std_dev_array = vxCreateArray(vxGetContext((vx_reference)_graph->get()), VX_TYPE_FLOAT32, _batch_size * 3);
+    vx_status mean_status = VX_SUCCESS;
+    mean_status |= vxAddArrayItems(_mean_array,_batch_size * 3, mean_vx.data(), sizeof(vx_float32));
+    mean_status |= vxAddArrayItems(_std_dev_array,_batch_size * 3, std_dev_vx.data(), sizeof(vx_float32));
+    _mirror.create_array(_graph ,VX_TYPE_UINT32, _batch_size);
+    if(mean_status != 0)
+        THROW(" vxAddArrayItems failed in the resize mirror normalize node (vxExtrppNode_ResizeMirrorNormalizeCropbatchPD    )  node: "+ TOSTR(mean_status) + "  "+ TOSTR(mean_status))
+    
+    unsigned int chnShift = 0;
+    vx_scalar chnToggle = vxCreateScalar(vxGetContext((vx_reference)_graph->get()), VX_TYPE_UINT32, &chnShift);
+
+    std::vector<uint32_t> dst_roi_width(_batch_size,_outputs[0]->info().width());
+    std::vector<uint32_t> dst_roi_height(_batch_size, _outputs[0]->info().height_single());
+
+    _dst_roi_width = vxCreateArray(vxGetContext((vx_reference)_graph->get()), VX_TYPE_UINT32, _batch_size);
```

Comment:
@SundarRajan28 Will these be tensors in the tensor PR instead of arrays?

Reply:
These will be tensors in the tensor PR